### PR TITLE
Problems with tinyMCE width and height, joomla 3.7

### PR DESCRIPTION
**Pull request by schnuti**
I think there are some confusions about the computed values of the width and height values for the tinyMCE editor. Personally I categorize this as a bug. 

In theory there could be a lot of default settings but this is more simple.

We get the values from user editable system wide parameters (with some funny results like height=750, the tinyMCE default )):
- tinyMCE editor default values
- tinyMCE editor settings
- this overrides the parameters from the form in all components and views.

Article edit
article_is

A second field in article edit as in PR 14520.
article_is2

Article frontend editing.
article frontend

The Joomla editor form field will never get any settings from external sources as default.

My propasal is to set the width hardcoded to 100% (or auto) and the height from:
- Joomla editor form field default settings
- the field's settings in the form field XML
- tinyMCE default setting in the code (if no value available). I'm not quite sure about this.
- override in a system plug in - onBeforeRender() method

Expected result
article_should

As tinyMCE automatically gets the height values from the hidden textarea field, we don't have to explicit set any values in the tinyMCE options. As well, we don't have to set any editor instance values as in #14520.

If you want a width per field, this has to be added to each editor instance as in PR #14520. As far as I could see there is no automatic loading as with the height.

In this PR I have removed the parameters in the tinyMCE editor, removed the tinyMCE options, added defaults in tinyMCE code and deprecated the language strings. I did not change the legacy part. Width is now overridable with a custom template/layout and height by using a custom system plugin - method onBeforerender().

Test instructions:
1. Add an extra editor field as in #14520 with height. The original article text field has no expicit value, default 500px height from the form field.

In the article form XML administrator/components/com_content/models/forms/article.xml add one more editor element (somewhere after line 
`<fieldset name="basic" label="COM_CONTENT_ATTRIBS_FIELDSET_LABEL">`)

`<field name="text2" type="editor" hide="menu,module,contact"
 label="Text2" description="" filter="JComponentHelper::filterText" buttons="true" height="250"/>`

2. Check the result without adding the patch. Play with the tinyMCE editor settings a bit. Go to the tab(s) advanced in tinyMCE and modify Width and Height to what a user could enter. e.g. width 50%, height=200.
Have a look to the new second field in article edit, backend categories and frontend article edit/creation as well. 

3. Apply this patch.
There are no settings so just reload the article edit page and .
Check and compare the results.

@Fedik  Is it easy to add the width if !100% as in PR 14520? The width could then be set in the form field. Needed? 

Related
#14917
#14835

### CODE REVIEW

File: administrator/language/en-GB/en-GB.plg_editors_tinymce.ini [hunk 1]
```diff
@@ -52,10 +52,6 @@ PLG_TINY_FIELD_FUNCTIONALITY_DESC="Select level of functionality."
 PLG_TINY_FIELD_FUNCTIONALITY_LABEL="Functionality"
 PLG_TINY_FIELD_HR_DESC="Show or hide the Horizontal Rule button."
 PLG_TINY_FIELD_HR_LABEL="Horizontal Rule"
-PLG_TINY_FIELD_HTMLHEIGHT_DESC="Height of HTML editor. Only applies in Advanced and Extended mode."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_editors_tinymce.ini]: Please don't remove the unused language strings. You can mark them as unused but the standard says we don't delete thrm

Comment (schnuti) [inline administrator/language/en-GB/en-GB.plg_editors_tinymce.ini]: OK! I was not sure. How do I mark them?

Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_editors_tinymce.ini]: Something like

`; The following  xxx strings are deprecated and will be removed with 4.0`

**Comment by Fedik** [comment]
>  Is it easy to add the width if !100% as in PR 14520?

sorry, not very understood your question,
14520 already merged

**Comment by schnuti** [comment]
@Fedik 
I wrote - **as in**. That is handle the width in the same way.

**Comment by AlexRed** [comment]
I have tested this item :white_check_mark: successfully on 122caac71e7604ec69920f725d3117c3195b9209<br /><br />Patch ok for me.
Thanks  :smiley: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15059">issues.joomla.org/tracker/joomla-cms/15059</a>.</sub>

**Comment by schnuti** [comment]
@AlexRed 
Thanks for testing.
Do you as I, think, it should go into 3.7.0? I guess it's harder to remove the parameters later on. It's late!

File: administrator/language/en-GB/en-GB.plg_editors_tinymce.ini (hunk 1 above)
Comment (schnuti) [inline administrator/language/en-GB/en-GB.plg_editors_tinymce.ini]: @brianteeman 
Thanks. Now I only have to find out how to revert the change and add the new line. I'm only using gitHub occasionally. Any hint?

**Comment by AlexRed** [comment]
yes, also I hope we can solve the tinyMCE width problem in 3.7.0 or 3.7.1

**Comment by schnuti** [comment]
@brianteeman  I've reverted the language change and added a remark

**Comment by schnuti** [comment]
I can't think of any situation where a width <> 100% (see images above) is needed. If you know one, I've now tested a solution using Fediks editor instances that works fine.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :red_circle: unsuccessfully on 8a2f3380966d648b8907aed70536d0b7b7af5897<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15059">issues.joomla.org/tracker/joomla-cms/15059</a>.</sub>

**Comment by schnuti** [comment]
@franz-wohlkoenig 
Could you please check if you have the JS file in the directory.
media/editors/tinymce/plugins/jdragdrop/plugin-min.js
It has obviously bin moved to media/editors/tinymce/js/plugins/dragdrop/plugin-min.js

As far as I can see this PR has nothing to do with this javascript loading error.  
Related? #15057

**Comment by franz-wohlkoenig** [comment]
>  media/editors/tinymce/plugins/jdragdrop/plugin-min.js

jdragdrop/plugin-min.js doesn't exist.

> media/editors/tinymce/js/plugins/dragdrop/plugin-min.js

Folder exist.

**Comment by schnuti** [comment]
@franz-wohlkoenig 
I think you somehow got out of sync.
see e.g.
https://github.com/joomla/joomla-cms/tree/staging/media/editors/tinymce/js/plugins/dragdrop

**Comment by franz-wohlkoenig** [comment]
@schnuti got same Path&File as https://github.com/joomla/joomla-cms/tree/staging/media/editors/tinymce/js/plugins/dragdrop

**Comment by schnuti** [comment]
@franz-wohlkoenig 
Do you really find it on your instance? Do you override the tinyMCE settings? (personally I love it!)
As I said "**This PR do not change anything regarding Javascript"**"

**Comment by franz-wohlkoenig** [comment]
> Do you really find it on your instance?

bildschirmfoto 2017-04-30 um 11 28 34

> Do you override the tinyMCE settings?

Clean Install of latest nightly Build. After applied PR got [Error described in Comment](https://github.com/joomla/joomla-cms/pull/15059#issuecomment-298215149)

**Comment by schnuti** [comment]
@franz-wohlkoenig 
That's as  far I go, I have my solution - override-. I do NOT need this patch.
By!

**Comment by schnuti** [comment]
@franz-wohlkoenig 
I have to apologize! There are no conflicts above - but there should be, 
My PR overrides  somehow #15057 that moves the js to an "external" path.
I get no differences with staging.
How can I fix that? New PR?

**Comment by franz-wohlkoenig** [comment]
@schnuti sorry, don't know how to fix, maybe @Bakual can help?

**Comment by schnuti** [comment]
As I understand this should not be possible.
As I said - when I compare with JoomlaCMS/staging I get " Able to merge." and an update is not possible!
My **guess** for now is that the PatchTester is the problem copying the old PHP file to the test system and that a commit to Joomla staging would work.
But I would need some advise.

**Comment by Bakual** [comment]
Yep, Patchtester can cause some funny results if the PR branch doesn't have conflicts but is outdated neverthless. Because it copies the files instead of merging the changes.
You can merge (or rebase) the staging branch into your PR branch to fix that issue.

**Comment by schnuti** [comment]
@Bakual 
Thanks for the hint. I found a way to merge the staging changes into the PR branch using the GitHub web interface.
@franz-wohlkoenig 
Could you give it another try. I've tested and it works for me using the PatchTester.
Thanks for testing and finding the problem in the first place!
